src/MotionPointInterpolator.cpp

```diff
@@ -0,0 +1,268 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Motion Point Interpolator
+//
+// Stores a queue of Motion Points commands from a topic. Interpolates
+// between points when a Motion Point is requested.
+//
+////////////////////////////////////////////////////////////////////////////
+
+// Associated header
+#include "iarc7_motion/MotionPointInterpolator.hpp"
+
+//System Headers
+#include <algorithm>
+
+using namespace Iarc7Motion;
+
+// Construct the object need a node handle to register
+// the subscriber for velocity targets
+MotionPointInterpolator::MotionPointInterpolator(ros::NodeHandle& nh) :
+motion_points_subscriber_(),
+motion_point_targets_()
+{
+    motion_points_subscriber_ = nh.subscribe(
+                    "motion_point_targets",
+                    100,
+                    &MotionPointInterpolator::processMotionPointArray,
+                    this);
+
+    // Put an initial time in the buffer so that buffer is never empty
+    MotionPointStamped zero_state;
+    zero_state.header.stamp = ros::Time::now();
+    motion_point_targets_.emplace_back(zero_state);
+}
+
+// Called by a class user to get a target motion point
+// Makes the call to trim the queue during the class
+// Interpolates if there is more than one twist
+void MotionPointInterpolator::getTargetMotionPoint(
+                const ros::Time& current_time,
+                MotionPointStamped& target_motion_point)
+{
+
+    // Trim velocity queue when done we will have one or two velocities available.
+    trimMotionPointQueue(motion_point_targets_, current_time);
+
+    if(motion_point_targets_.size() == 1)
+    {
```

Comment:
Picky, but in our codebase, I see the `{` on the same line as the if statement and on the next line. Do we have a convention and can we adopt it universally? Personally I prefer same line.

Reply:
The convention is next line but we decided to not pick on it code reviews and just fix it when we see it.

Reply:
The convention we decided on is `{` on the same line, also convention says you need a space between `if` and `(`.  We're using the Google C++ style guide with some exceptions, but this isn't one of the exceptions.  However, I don't care, and I'd say Levi can fix it if he feels like it.
https://google.github.io/styleguide/cppguide.html#Conditionals

Reply:
Can't believe that I get this wrong so often.